src/components/container/AppleHealthRecordsSync/AppleHealthRecordsSync.tsx

```diff
@@ -0,0 +1,124 @@
+
+    const [platform, setPlatform] = useState<string>();
+    const [status, setStatus] = useState<AppleHealthRecordsStatus>();
+    const [hasData, setHasData] = useState<boolean>(false);
+    const [connecting, setConnecting] = useState<boolean>(false);
+    const [showHelp, setShowHelp] = useState<boolean>(false);
+
+    const isConnecting = useRef<boolean>(false);
+    const recheckTimeoutId = useRef<ReturnType<typeof setTimeout>>();
+
+    const applyPreviewState = (previewState: AppleHealthRecordsSyncPreviewState) => {
+        setPlatform(previewState === 'wrong platform' ? 'Web' : 'iOS');
+        setStatus(previewState.startsWith('enabled') ? 'enabled' : previewState === 'disabled' ? 'disabled' : undefined);
+        setHasData(previewState === 'enabled with data');
+        setConnecting(previewState === 'enabled no data yet');
+        setShowHelp(false);
+    };
+
+    useInitializeView(() => {
+        if (props.previewState) {
+            applyPreviewState(props.previewState);
+            return;
+        }
+
+        clearTimeout(recheckTimeoutId.current);
+
+        if (!platform) {
+            MyDataHelps.getDeviceInfo().then(deviceInfo => {
+                setPlatform(deviceInfo.platform);
+            });
+            return;
+        }
+
+        if (platform === 'iOS') {
+            MyDataHelps.getDataCollectionSettings().then(settings => {
+                setStatus(settings.appleHealthRecordsEnabled ? 'enabled' : 'disabled');
+                if (settings.appleHealthRecordsEnabled) {
+                    const checkDataAvailability = (recheckAttempts: number = 0) => {
+                        MyDataHelps.getDataAvailability().then(dataAvailability => {
+                            setHasData(dataAvailability.appleHealthRecords);
+                            if (dataAvailability.appleHealthRecords || recheckAttempts === MAX_DATA_AVAILABILITY_RECHECK_ATTEMPTS) {
+                                setConnecting(false);
+                            } else if (isConnecting.current) {
+                                recheckTimeoutId.current = setTimeout(() => checkDataAvailability(recheckAttempts + 1), 5000);
+                            }
+                        });
+                    };
+                    checkDataAvailability();
+                } else {
+                    setHasData(false);
+                    setConnecting(false);
+                }
+            });
+        }
+    }, [], [props.previewState, platform], () => clearTimeout(recheckTimeoutId.current));
+
+    useEffect(() => {
+        isConnecting.current = connecting;
+    }, [connecting]);
+
+    if (!status || (props.showWhen && props.showWhen !== status)) {
+        return null;
+    }
+
+    const getSubtitle = () => {
+        if (status === 'enabled') {
+            return hasData ? language('connected') : connecting ? language('syncing-data') : language('no-data');
+        }
+        return language('apple-health-records-sync-prompt');
```

Comment:
Readability could be improved a bit here.
```suggestion
        if (status !== 'enabled') {
            return language('apple-health-records-sync-prompt');
        }
        if (hasData) {
            language('connected');
        }
        return connecting ? language('syncing-data') : language('no-data');
```

Reply:
Done.